### PR TITLE
[POAE7-2904] fix some bugs when enable partial_agg_pattern

### PR DESCRIPTION
**Pull request by harborn**

### CODE REVIEW

File: cpp/src/cider-velox/test/CiderOperatorTest.cpp [hunk 1]
```diff
@@ -444,10 +478,7 @@ TEST_F(CiderOperatorTest, partial_avg_null) {
                        .planNode();
 
   auto duckdbSql = "SELECT row(null, 0)";
-  // FIXME: For partial avg, duckdb returns a row (null, 0) while velox returns a null row
-  // when input is an all null column.
```
Comment (yma11) [inline cpp/src/cider-velox/test/CiderOperatorTest.cpp:448]: Don't remove above comments which are useful.

**Review by chaojun-zhang — DISMISSED** [review]

File: cpp/src/cider-velox/src/ciderTransformer/CiderPlanTransformerOptions.cpp [hunk 2]
```diff
@@ -25,6 +25,6 @@ DEFINE_bool(left_deep_join_pattern, false, "Enable LeftDeepJoinPattern ");
 DEFINE_bool(compound_pattern, false, "Enable CompoundPattern ");
 DEFINE_bool(filter_pattern, true, "Enable FilterPattern ");
 DEFINE_bool(project_pattern, true, "Enable ProjectPattern ");
-DEFINE_bool(partial_agg_pattern, false, "Enable PartialAggPattern ");
+DEFINE_bool(partial_agg_pattern, true, "Enable PartialAggPattern ");
```
Comment (winningsix) [inline cpp/src/cider-velox/src/ciderTransformer/CiderPlanTransformerOptions.cpp:28]: default false?

File: cpp/src/cider-velox/test/CiderOperatorTest.cpp [hunk 3]
```diff
@@ -309,11 +339,14 @@ TEST_F(CiderOperatorTest, aggOnExpr_withoutCond) {
       " sum(cast(l_linenumber as bigint)) as sum_linenumber,"
       " count(l_linenumber) as cnt "
       " from tmp";
+
+  assertQuery(veloxPlan, duckDbSql);
   assertQuery(resultPtr, duckDbSql);
 }
-
+#if 0
```
Comment (winningsix) [inline cpp/src/cider-velox/test/CiderOperatorTest.cpp:346]: skipTests?

**Review by winningsix — COMMENTED** [review]
Just leave some minor comments.

**Review by yma11 — APPROVED** [review]

**Review by yma11 — APPROVED** [review]

**Review by winningsix — APPROVED** [review]

**Pull request merged**